scalafix-reflect/src/main/scala/scalafix/internal/reflect/ScalafixToolbox.scala

```diff
@@ -63,9 +63,19 @@ class Compiler() {
   settings.outputDirs.setSingleOutput(target)
   getClass.getClassLoader match {
     case u: URLClassLoader =>
-      settings.classpath.value = u.getURLs
-        .map(x => URLDecoder.decode(x.getPath, "UTF-8"))
-        .mkString(File.pathSeparator)
+      val paths = u.getURLs.toList.map(u => {
+        if (u.getProtocol.startsWith("bootstrap")) {
+          import java.io._
+          import java.nio.file._
+          val stream = u.openStream
+          val tmp = File.createTempFile("bootstrap-" + u.getPath, ".jar")
+          Files.copy(stream, Paths.get(tmp.getAbsolutePath), StandardCopyOption.REPLACE_EXISTING)
+          tmp.getAbsolutePath
+        } else {
+          URLDecoder.decode(u.getPath, "UTF-8")
+        }
+      })
+      settings.classpath.value = paths.mkString(File.pathSeparator)
```

Comment:
I was thinking, is it possible to pass in a `Classloader` instead of `classpath: String`?

Reply:
Nope.

The Scala compiler needs to traverse the entire classpath in order to correctly load symbols. A generic `ClassLoader` doesn't provide this functionality.

We have tried this in `ReflectGlobal` that takes a `ClassLoader` (it's used inside `Toolbox` which is part of our runtime reflection framework) and ran into hard- or impossible-to-fix problems on pretty common Scala code. After spending some time trying to fix those problems, we decided to abandon that direction of research.

Reply:
I take your word for it! It's a shame.